File: zbus-api/zbus-api-nodejs/examples/rpc.js
var zbus = require("../zbus");
var Message = zbus.Message;
var RemotingClient = zbus.RemotingClient;
var Rpc = zbus.Rpc;


var client = new RemotingClient("127.0.0.1:15555");
client.connect(function(){
    var rpc = new Rpc(client, "MyRpc");
    rpc.module = "Interface";
    rpc.invoke({
            method: "plus",
            params:[1,2]
        },
        function(res){ // res.result, or result.stackTrace
            console.log(res.result); 
        }
    );
});

File: api2/csharp/Zbus.Examples/Rpc/Rpc.cs
using Zbus.Broker;
using Zbus.RPC;
using System;
namespace Zbus.Examples
{
   class RpcClientTest
   {
      public static void Main(string[] args)
      {
         IBroker broker = new ZbusBroker();
         Rpc rpc = new Rpc(broker);

         Request req = new Request();
         req.Mq = "MyRpc";
         req.Module = "MyService";
         req.Method = "getString";
         req.Params = new object[] { "test" };


         object res = rpc.Invoke(req);
         Console.WriteLine(res);

         broker.Dispose();
         Console.ReadKey();
      }
   }
}
